src/main/resources/jshint-shell.js

```diff
@@ -28,6 +28,13 @@
     var OPTIONS_ARG = 4;
 
     var options = JSON.parse(stripJsonComments(args[OPTIONS_ARG]));
+    var globals;
+
+    if (options.globals) {
+        globals = options.globals;
+        delete options.globals;
```

Comment:
Apparently, delete just removes the property the object.  It doesn't have anything to do with memory.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete